### ISSUE TITLE
JinjaRule.<bound method AnsibleLintRule.matchtasks of jinja: Rule that looks inside jinja2 templates.>

### ISSUE THREAD

**Issue by huguesjoyal** [ansible/ansible-lint#2793]
##### Summary

ansible-lint return the following error :

JinjaRule.<bound method AnsibleLintRule.matchtasks of jinja: Rule that looks inside jinja2 templates.>

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT
```console
ansible-lint --version
ansible-lint 6.9.1 using ansible 2.14.0

MacOS 12.6.1
Python 3.10.8
ansible installation method: pip3
ansible-lint installation method: pip3
```

##### STEPS TO REPRODUCE

The following role produce the error :
```yml
---
- name: Failed test
  ansible.builtin.set_fact:
    certbot_plugins: "{{ [certbot_plugin] + (certbot_certificates | json_query(certbot_plugins_query)) | unique }}"
  vars:
    certbot_plugins_query: "[].plugin"
````

The problem doesn't occur when I do it in two step:
```yml
- name: Passed test in two step
  ansible.builtin.set_fact:
    certbot_plugins: "{{ certbot_certificates | json_query(certbot_plugins_query) }}"
  vars:
    certbot_plugins_query: "[].plugin"

- name: Passed test in two step
  ansible.builtin.set_fact:
    certbot_plugins: "{{ ([certbot_plugin] + certbot_plugins) | unique }}"
```

##### Desired Behavior

Lint should not be throwing an error.

##### Actual Behavior

The following warning is thrown
```paste
WARNING  Ignored exception from JinjaRule.<bound method AnsibleLintRule.matchtasks of jinja: Rule that looks inside jinja2 templates.> while processing <file> (tasks): 'TypeError' object has no attribute 'message'
```

**Comment by bluikko** [comment]
I have a very similar warning message but it is `KeyError` instead of `TypeError`, I wonder if this is about same issue?
```yaml
---
- name: Set host address
  ansible.builtin.set_fact:
    database_host: "{{ query('ansible.builtin.inventory_hostnames', 'database') | map('extract', hostvars, ['ansible_facts', 'eth0', 'ipv4', 'address']) }}"
```
Produces warning:
```console
WARNING  Ignored exception from JinjaRule.<bound method AnsibleLintRule.matchtasks of jinja: Rule that looks inside jinja2 templates.> while processing roles/database/tasks/database.yml (tasks): 'KeyError' object has no attribute 'message'
```
